pkg/kvstore/backend.go

```diff
@@ -163,6 +163,10 @@ type BackendOperations interface {
 	// Update atomically creates a key or fails if it already exists
 	Update(ctx context.Context, key string, value []byte, lease bool) error
 
+	// UpdateIfDifferent atomically updates a key but only if the value is
```

Comment:
Is atomic property provided by the implementations? I am not saying it should be, but I don't think that calling these updates "atomic" is correct, as there are races possible because we don't lock the keys in any way.

Reply:
Fixed